### ISSUE TITLE
Paperclip closing, time to move to ActiveStorage?

### ISSUE THREAD

**Issue by askl56** [solidusio/solidus#2725]
```Seems that Thoughtbot guys decided to retire Paperclip and are stopping all further development and support for it.

A handful of PRs that are marked "merge before deprecating": https://github.com/thoughtbot/paperclip/pulls 

Can't really find announcement or discussion around it though.
```

Is there a migration plan for moving from Solidus to Paperclip?

**Comment by askl56** [comment]
Anyone on this?

**Comment by gmacdougall** [comment]
We are planning on migration away from Paperclip. There was some initial work done in #2328 but it needs some additional work.

**Comment by swcraig** [comment]
The work in #2337 most likely needs to be merged before we are able to easily extract Paperclip into an extension.

**Comment by askl56** [comment]
@gmacdougall Happy to help in any way I can

**Comment by exocode** [comment]
Hi there, any news on this topic. 

Came here by a research for Activestorage. I saw the team on Spree did an update: They already implemented ActiveStorage.

My questionis if there is a near future where this feature will be migrated?

Thank you very much

**Comment by jacobherrington** [comment]
@exocode It looks like @swcraig is continuing work on this change right now!

**Comment by askl56** [comment]
Any update on this?

**Comment by equivalent** [comment]
@askl56  from https://spreecommerce.org/spree-3-5-and-3-6-with-rails-5-2-ruby-2-5-and-activestorage-support-released/

```
ActiveStorage
Active Storage facilitates uploading files to a cloud storage service like Amazon S3, Google Cloud Storage, or Microsoft Azure Storage and attaching those files to Active Record objects. It comes with a local disk-based service for development and testing and supports mirroring files to subordinate services for backups and migrations.

ActiveStorage was introduced in Rails 5.2 and with Spree 3.6 release it’s the default file storage option. We’ll still support Paperclip until Spree 4.0 when it will be removed from Spree core.
```

so I guess it's in

**Comment by swcraig** [comment]
I am no longer actively working on this.
@elia has graciously picked up the work [here](https://github.com/solidusio/solidus/pull/2974).

**Comment by kennyadsl** [comment]
This is now merged in master. It will be available when Rails 6.1 will be released.
